2018/11/05/Makefile

```diff
@@ -1,10 +1,19 @@
-testingmlp: testingmlp.cpp
-	c++ -O2 -o testingmlp testingmlp.cpp
+CXXFLAGS = -O2 -std=c++11
 
+testingmlp: testingmlp.o generated.o
+	$(CXX) $(CXXFLAGS) -o $@ $^
 
 
-test: testingmlp.cpp
+ifneq ($(GENERATE), 0)
+generated.cpp : gen.py
```

Comment:
The idea here is to generate the `generated.cpp` file in during the build, so you dont' have to manually fix up any code. The `generated.cpp` file is still checked in to git, however, some user doesn't have `python3` they can just run `make GENERATE=0` and use the existing file (of course they can't generate but if they just want to build that's fine).